File: backend/src/cmd/record.rs
use crate::common::entity::record::RollcallRecord;
use crate::database::database_pool;
use crate::service::record_service;

/// 列出最近的点名记录
#[tauri::command]
pub async fn record_list() -> Result<Vec<RollcallRecord>, String> {
    let rb = database_pool::database().await.map_err(|e| e.to_string())?;
    record_service::list_all(&*rb).await.map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn record_batch_update_attendance_status(ids: Vec<i64>, attendance_status: i8) -> Result<Vec<RollcallRecord>, String> {
    let rb = database_pool::database().await.map_err(|e| e.to_string())?;
    record_service::update_attendance_status(&*rb, ids, attendance_status)
        .await
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn record_batch_update_remark(ids: Vec<i64>, remark: String) -> Result<Vec<RollcallRecord>, String> {
    let rb = database_pool::database().await.map_err(|e| e.to_string())?;
    record_service::update_remark(&*rb, ids, remark)
        .await
        .map_err(|e| e.to_string())
}

File: frontend/src/lib/commands/record.ts
import {invoke} from "@tauri-apps/api/core";
import {RollcallRecord} from "$types/RollcallRecord";


export async function list() {
  return await invoke<RollcallRecord[]>("record_list");
}

export async function update(ids: bigint[], attendance_status: number, remark: string) {
  return await invoke<RollcallRecord[]>("record_batch_update", {
    ids: ids,
    attendanceStatus: attendance_status,
    remark: remark
  })
}

export async function update_attendance_status(ids: bigint[], attendance_status: number) {
  return await invoke<RollcallRecord[]>("record_batch_update_attendance_status", {
    ids: ids,
    attendanceStatus: attendance_status
  })
}

export async function update_remark(ids: bigint[], remark: string) {
  return await invoke<RollcallRecord[]>("record_batch_update_remark", {
    ids: ids,
    remark: remark
  })
}
